crates/bevy_math/src/bounding/bounded2d/primitive_impls.rs

```diff
@@ -0,0 +1,457 @@
+    fn aabb_2d(&self, translation: Vec2, rotation: f32) -> Aabb2d {
+        let normal = rotate_vec2(*self.normal, rotation);
+        let facing_x = normal == Vec2::X || normal == Vec2::NEG_X;
+        let facing_y = normal == Vec2::Y || normal == Vec2::NEG_Y;
+
+        // Dividing `f32::MAX` by 2.0 can actually be good so that we can do operations
+        // like growing or shrinking the AABB without breaking things.
+        let half_width = if facing_x { 0.0 } else { f32::MAX / 2.0 };
+        let half_height = if facing_y { 0.0 } else { f32::MAX / 2.0 };
+        let half_size = Vec2::new(half_width, half_height);
+
+        Aabb2d {
+            min: translation - half_size,
+            max: translation + half_size,
+        }
+    }
+
+    fn bounding_circle(&self, translation: Vec2, _rotation: f32) -> BoundingCircle {
+        BoundingCircle::new(translation, f32::MAX / 2.0)
+    }
+}
+
+impl Bounded2d for Line2d {
+    fn aabb_2d(&self, translation: Vec2, rotation: f32) -> Aabb2d {
+        let direction = rotate_vec2(*self.direction, rotation);
+
+        // Dividing `f32::MAX` by 2.0 can actually be good so that we can do operations
+        // like growing or shrinking the AABB without breaking things.
+        let max = f32::MAX / 2.0;
+        let half_width = if direction.x == 0.0 { 0.0 } else { max };
+        let half_height = if direction.y == 0.0 { 0.0 } else { max };
+        let half_size = Vec2::new(half_width, half_height);
+
+        Aabb2d {
+            min: translation - half_size,
+            max: translation + half_size,
+        }
+    }
+
+    fn bounding_circle(&self, translation: Vec2, _rotation: f32) -> BoundingCircle {
+        BoundingCircle::new(translation, f32::MAX / 2.0)
+    }
+}
+
+impl Bounded2d for Segment2d {
+    fn aabb_2d(&self, translation: Vec2, rotation: f32) -> Aabb2d {
+        // Rotate the segment by `rotation`
+        let direction = Direction2d::from_normalized(rotate_vec2(*self.direction, rotation));
+        let segment = Self { direction, ..*self };
+        let (point1, point2) = (segment.point1(), segment.point2());
+
+        Aabb2d {
+            min: translation + point1.min(point2),
+            max: translation + point1.max(point2),
+        }
+    }
+
+    fn bounding_circle(&self, translation: Vec2, _rotation: f32) -> BoundingCircle {
+        BoundingCircle::new(translation, self.half_length)
+    }
+}
+
+impl<const N: usize> Bounded2d for Polyline2d<N> {
+    fn aabb_2d(&self, translation: Vec2, rotation: f32) -> Aabb2d {
+        Aabb2d::from_point_cloud(translation, rotation, self.vertices)
+    }
+
+    fn bounding_circle(&self, translation: Vec2, rotation: f32) -> BoundingCircle {
+        BoundingCircle::from_point_cloud(translation, rotation, &self.vertices)
+    }
+}
+
+impl Bounded2d for BoxedPolyline2d {
+    fn aabb_2d(&self, translation: Vec2, rotation: f32) -> Aabb2d {
+        Aabb2d::from_point_cloud(translation, rotation, self.vertices.to_vec())
```

Comment:
Can `Aabb2d::from_point_cloud` just take a slice to not allocate here (and in others like this)?